tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -7,12 +7,256 @@
+
+        assert result.E7403131 == 2
+        assert result.E7403132 == 60
+
+    def test_flaechen_one_under_60_should_set_fields(self):
+        flaechen = [59, 60]
+
+        result = EWohnUnter60(flaechen)
+
+        assert result.E7403131 == 1
+        assert result.E7403132 == 59
+
+
+class TestEWohn60bis100:
+    def test_flaeche_60_should_set_fields(self):
+        flaechen = [60]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 60
+
+    def test_flaeche_99_should_set_fields(self):
+        flaechen = [99]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 99
+
+    def test_flaeche_100_should_retain_zeroes(self):
+        flaechen = [100]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 0
+        assert result.E7403142 == 0
+
+    def test_flaechen_under_100_should_set_fields(self):
+        flaechen = [99, 60]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 2
+        assert result.E7403142 == 159
+
+    def test_flaechen_one_under_100_should_set_fields(self):
+        flaechen = [99, 100]
+
+        result = EWohn60bis100(flaechen)
+
+        assert result.E7403141 == 1
+        assert result.E7403142 == 99
+
+
+class TestEWohnAb100:
+    def test_flaeche_100_should_set_fields(self):
+        flaechen = [100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 1
+        assert result.E7403152 == 100
+
+    def test_flaeche_99_should_retain_zeroes(self):
+        flaechen = [99]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 0
+        assert result.E7403152 == 0
+
+    def test_flaechen_from_100_should_set_fields(self):
+        flaechen = [100, 100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 2
+        assert result.E7403152 == 200
+
+    def test_flaechen_one_under_100_should_set_fields(self):
+        flaechen = [99, 100]
+
+        result = EWohnAb100(flaechen)
+
+        assert result.E7403151 == 1
+        assert result.E7403152 == 100
+
+
+class TestEWeitereWohn:
+    def test_should_construct_from_dict_correctly(self):
+        weitere_wohnraeume = WeitereWohnraeumeDetails.parse_obj({"anzahl": 2, "flaeche": 42})
+
+        result = EWeitereWohn(weitere_wohnraeume)
+
+        assert result.E7403121 == 2
+        assert result.E7403122 == 42
+
+
+class TestEGaragen:
+    def test_should_construct_from_dict_correctly(self):
+        garagen_anzahl = GaragenAnzahl.parse_obj({"anzahl_garagen": 2})
+
+        result = EGaragen(garagen_anzahl)
+
+        assert result.E7403171 == 2
+
+
+class TestEAngDurchschn:
+    def test_wohnflaeche_under_60_should_set_others_to_none(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(59).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Wohn_Unter60 is not None
```

Comment:
Should we maybe instead test then `assert result.Wohn_Unter60 == EWohnUnter60(gebaeude.wohnflaeche)` or similar?

Reply:
I purposefully do not test for the contents of the object here since that part is tested in `class TestEWohnUnter60`. That way, testing the calculation of the contents of Wohn_Unter60 instance is separated from its assignment in EAngDurchschn, where only the latter is a concern of this test. Does that make sense?

Reply:
It does make sense, but I still am not sure that it's set correctly. I just think it doesn't hurt to explicitly test the content and it'd give me a higher confidence.

Reply:
Done.